isic/zip_download/api.py

```diff
@@ -32,8 +34,18 @@ def get_attributions(attributions: Iterable[str]) -> list[str]:
     return [x[0] for x in attributions]
 
 
-def get_zip_download_token(token: str) -> dict:
-    return TimestampSigner().unsign_object(token, max_age=timedelta(days=1))
+class ZipDownloadTokenAuth(APIKeyQuery):
+    param_name = "token"
+
+    def authenticate(self, request: HttpRequest, key: str) -> dict:
+        try:
+            token_dict = TimestampSigner().unsign_object(key, max_age=timedelta(days=1))
+        except BadSignature:
+            logger.exception("Bad zip download token passed")
+            raise AuthenticationError
+
+        token_dict["token"] = key
```

Comment:
We could remove this if we refactored `zip_file_listing` to re-sign the token object internally, wasn't sure which you'd prefer.

Reply:
No, this looks good to me.